src/core/annotation.js

```diff
@@ -987,6 +987,22 @@ class WidgetAnnotation extends Annotation {
     const vPadding = defaultPadding + Math.abs(descent) * fontSize;
     const defaultAppearance = this.data.defaultAppearance;
 
+    if (this.data.comb) {
+      const combWidth = (totalWidth / this.data.maxLen).toFixed(2);
```

Comment:
To prevent this method from being too big, I would suggest to move the comb handling code to a private method named `_renderCombText` similar to what you did for the multiline strings.